### PR TITLE
Refine add_test and Add WORKING_DIRECTORY

### PR DESCRIPTION
**Pull request by Aurelius84**
+ Refine redundant code of `ADD_TEST` by `foreach`
+ Add `WORKING_DIRECTORY ${CMAKE_BINARY_DIR}`

### CODE REVIEW

File: python/CMakeLists.txt [hunk 1]
```diff
@@ -12,70 +12,53 @@ ADD_CUSTOM_COMMAND(OUTPUT ${CINN_CORE_API} POST_BUILD
...
+    test_pe_elementwise
+    test_pe_reduction
+    test_pe_transform
+    test_op_nn
+    test_op_broadcast
 )
 
-ADD_TEST(NAME test_cinn_pe_elementwise
+foreach(basic_test_name ${BASIC_TEST_NAMES})
+    string(REGEX REPLACE "/test" "/test_cinn" ${ctest_name} ${basic_test_name})
```
Comment (Joejiong) [inline python/CMakeLists.txt]: foreach(basic_test_name ${BASIC_TEST_NAMES})
    string(REGEX REPLACE "/test" "/test_cinn" ${ctest_name} ${basic_test_name})  string(REGEX REPLACE 好像要加个“${ctest_name}”，“${basic_test_name}”
https://gitlab.kitware.com/cmake/cmake/-/issues/18737

Comment (Aurelius84) [reply python/CMakeLists.txt]: Thanks, done

**Review by Joejiong — APPROVED** [review]
LGTM

**Comment by haozech** [comment]
LGTM

**Review by Superjomn — APPROVED** [review]
LGTM

**Pull request merged**